### PR TITLE
smb_ntlm: do not send lm & ntlm to server for guest login

### PR DESCRIPTION
**Pull request by Kxuan**
I do a quick fix for #112. It is not a prefect patch, but it works for me.

@jbkempf

### CODE REVIEW

**Comment by jbkempf** [comment]
Seems nice.

@chouquette an opinion?

File: src/smb_ntlm.c [hunk 1]
```diff
@@ -382,9 +381,15 @@ void        smb_ntlmssp_response(uint64_t srv_challenge, uint64_t srv_ts,
     auth->flags = 0x60088215;
 
 
-    __AUTH_APPEND(lm, lm2, 24, cursor)
-    __AUTH_APPEND(ntlm, ntlm2, blob_size + 16, cursor)
-
+    if (!*domain && !*user && !*password) {
+        utf_sz = smb_to_utf16("", 0, &utf);
```
Comment (chouquette) [inline src/smb_ntlm.c]: Are you sure this is required? As far as I can see smb_tu_utf16 will set `utf` to NULL when the size parameter is 0

File: src/smb_ntlm.c [hunk 2]
```diff
@@ -47,9 +47,8 @@
 
 #ifdef HAVE_BSD_STRING_H
 #include <bsd/string.h>
-#else
-#include "compat.h"
 #endif
+#include "compat.h"
```
Comment (chouquette) [inline src/smb_ntlm.c]: This seems unrequired

Comment (Kxuan) [inline src/smb_ntlm.c]: closesocket is provided with compact.h. we need this macro whether or not the system have bsd/string.h.

File: src/smb_ntlm.c (hunk 1 above)
Comment (Kxuan) [inline src/smb_ntlm.c]: sure. but it works

Comment (chouquette) [inline src/smb_ntlm.c]: Yeah but it seems you could remote the call to smb_to_utf16 & free(utf) while keeping the same functionality, no?

File: src/smb_ntlm.c (hunk 2 above)
Comment (chouquette) [inline src/smb_ntlm.c]: Oh this is fixing the build on some platform?

File: src/smb_ntlm.c (hunk 1 above)
Comment (Kxuan) [inline src/smb_ntlm.c]: yes, you are right. actually, there are no need to do the hash function to get smb_lm2_response and smb_ntlm2_response. should I remove it ?

Comment (chouquette) [inline src/smb_ntlm.c]: If you mean directely passing NULL to __AUTH_APPEND, then yes please!

Comment (Kxuan) [inline src/smb_ntlm.c]: ok I will do this on my next vacation.

**Comment by chouquette** [comment]
I allowed myself to amend the commit and pushed it, thanks for the bugreport, investigations, and the patch!

**Pull request closed without merging**